### PR TITLE
Add blog entry for lxboot x86

### PR DESCRIPTION
**Pull request by Mihnea0Firoiu**
The first blog post for lxboot x86

### CODE REVIEW

File: content/blog/2024-06-18-gsoc-lxboot_x86.mdx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+### Project overview
+
+The Linux boot protocol plays an important role in the initialization of the Linux operating system, emphasizing the importance of system optimization and scalability.
+This system ensures a seamless transfer of control from the bootloader to the Linux kernel by providing essential information including memory map, command line parameters, and device specifications.
+By adhering to a uniform boot protocol, Linux achieves a level of versatility across the hardware spectrum of architectures and bootloader functions.
+Essentially, this protocol is a commitment to a standardized approach that is not only stable but also facilitates bootloader development, improving compatibility and reliability.
+In particular, the Linux boot protocol greatly contributes to the overall robustness and reliability of the bootstrapping process on Linux-based systems. Almost, if not all, widely
+ available bootloaders that exist on the market are able to speak this protocol and Unikraft, being unable to be detected by these bootloaders as a Linux Boot Protocol compliant system,
+ is hindering the ability of Unikraft to boot on a very large amount of platforms.
+
```
Comment (StefanJum) [inline content/blog/2024-06-18-gsoc-lxboot_x86.mdx:28]:
```suggestion
In particular, the Linux boot protocol greatly contributes to the overall robustness and reliability of the bootstrapping process on Linux-based systems.
Almost, if not all, widely available bootloaders that exist on the market are able to speak this protocol and Unikraft, being unable to be detected by these bootloaders as a Linux Boot Protocol compliant system, is hindering the ability of Unikraft to boot on a very large amount of platforms.

```
Please use one line per sentence and one sentence per line

File: content/blog/2024-06-18-gsoc-lxboot_x86.mdx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+### Current Progress
+
+I already compiled `qemu 9.0.1` using the following flags: `--enable-debug --extra-cflags="-g3" --extra-ldflags="-g3" --disable-strip --enable-kvm --enable-avx2 --enable-avx512bw --enable-avx512f` to help me debug.
+At the moment, I am looking into creating `support/scripts/mklinux_x86.py` using as inspiration the already existing `mklinux.py` and the [ARM64 PR](https://github.com/unikraft/unikraft/pull/988).
+
+### Next Steps
+
+Once everything works for QEMU, I will move onto GRUB2 and look into making lxboot x86 work with it as ISO and as Disk.
```
Comment (StefanJum) [inline content/blog/2024-06-18-gsoc-lxboot_x86.mdx:36]: Try to add a few more detailed steps on what needs to happen in order for this to run on `qemu`. What the `mklinux_x86.py` script will contain, what the header of the image will look like, etc.

**Review by StefanJum — CHANGES_REQUESTED** [review]

**Review by mogasergiu — APPROVED** [review]
Other than what @StefanJum said, LGTM.

File: content/blog/2024-06-18-unikraft-gsoc-lxboot_x86.mdx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+| 0258/8      | 2.10+       | pref_address        | Preferred loading address                              |
+| 0260/4      | 2.10+       | init_size           | Linear memory required during initialization           |
+| 0264/4      | 2.11+       | handover_offset     | Offset of handover entry point                         |
+
+For more details about certain fields, read [here](https://www.kernel.org/doc/html/v5.4/x86/boot.html).
+
+### Next Steps
+
+Once everything works for QEMU, I will move onto GRUB2 and look into making lxboot x86 work with it as ISO and as Disk.
+One difference between two different boot loaders is that some have an identifier.
```
Comment (StefanJum) [inline content/blog/2024-06-18-unikraft-gsoc-lxboot_x86.mdx:82]: This sentence is a bit confusing. Would something like "Boot loaders have unique reserved identifiers, placed in the `type_of_loader` field" make more sense?

File: content/blog/2024-06-18-unikraft-gsoc-lxboot_x86.mdx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+Essentially, this protocol is a commitment to a standardized approach that is not only stable but also facilitates bootloader development, improving compatibility and reliability.
+In particular, the Linux boot protocol greatly contributes to the overall robustness and reliability of the bootstrapping process on Linux-based systems.
+Almost, if not all, widely available bootloaders that exist on the market are able to speak this protocol and Unikraft, being unable to be detected by these bootloaders as a Linux Boot Protocol compliant system, is hindering the ability of Unikraft to boot on a very large amount of platforms.
+
+### Current Progress
+
+I already compiled `qemu 9.0.1` using the following flags: `--enable-debug --extra-cflags="-g3" --extra-ldflags="-g3" --disable-strip --enable-kvm --enable-avx2 --enable-avx512bw --enable-avx512f` to help me debug.
+At the moment, I am looking into creating `support/scripts/mklinux_x86.py` using as inspiration the already existing `mklinux.py` and the [ARM64 PR](https://github.com/unikraft/unikraft/pull/988).
```
Comment (StefanJum) [inline content/blog/2024-06-18-unikraft-gsoc-lxboot_x86.mdx:31]:
```suggestion
At the moment, I am looking into creating `support/scripts/mklinux_x86.py` using as inspiration the already existing [`mklinux.py`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/mklinux.py) and the [ARM64 PR](https://github.com/unikraft/unikraft/pull/988).
```

File: content/blog/2024-06-18-unikraft-gsoc-lxboot_x86.mdx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+### Project overview
+
+The Linux boot protocol plays an important role in the initialization of the Linux operating system, emphasizing the importance of system optimization and scalability.
+This system ensures a seamless transfer of control from the bootloader to the Linux kernel by providing essential information including memory map, command line parameters, and device specifications.
+By adhering to a uniform boot protocol, Linux achieves a level of versatility across the hardware spectrum of architectures and bootloader functions.
+Essentially, this protocol is a commitment to a standardized approach that is not only stable but also facilitates bootloader development, improving compatibility and reliability.
+In particular, the Linux boot protocol greatly contributes to the overall robustness and reliability of the bootstrapping process on Linux-based systems.
+Almost, if not all, widely available bootloaders that exist on the market are able to speak this protocol and Unikraft, being unable to be detected by these bootloaders as a Linux Boot Protocol compliant system, is hindering the ability of Unikraft to boot on a very large amount of platforms.
```
Comment (StefanJum) [inline content/blog/2024-06-18-unikraft-gsoc-lxboot_x86.mdx:26]:
```suggestion
Almost, if not all, widely available bootloaders that exist on the market are able to speak this protocol and Unikraft, being unable to be detected by these bootloaders as a Linux Boot Protocol compliant system, is hindering the ability to boot on a very large amount of platforms.
```

**Review by StefanJum — CHANGES_REQUESTED** [review]
Few more comments.

**Review by StefanJum — APPROVED** [review]
Looks good now, thanks.

**Pull request merged**